src/toolkit/matl_sell_policy.cc

```diff
@@ -173,30 +197,37 @@ std::set<BidPortfolio<Material>::Ptr> MatlSellPolicy::GetMatlBids(
       req = *rit;
       qty = std::min(req->target()->quantity(), limit);
       bid_qty = excl ? quantize_ : package_->GetFillMass(qty);
-      if (bid_qty == 0) {
-        nbids = 0;
-      } else {
-        nbids = static_cast<int>(std::floor(qty / bid_qty));
+      if (bid_qty != 0) {
+        n_full_bids = static_cast<int>(std::floor(qty / bid_qty));
+        bids.assign(n_full_bids, bid_qty);
+
+        remaining_qty = fmod(qty, bid_qty);
+        if (!excl && (remaining_qty > 0) && 
+        (remaining_qty >= package_->fill_min())) {
+          // leftover material is enough to fill one more partial package. Add
+          // to bids
+          bids.push_back(remaining_qty);
+        }
+      }
+      // check transportability
+      int shippable_pkgs = transport_unit_->MaxShippablePackages(bids.size());
+      if (shippable_pkgs < bids.size()) {
+        // can't ship all bids. Pop the extras.
+        for (int i=0; i<(bids.size() - shippable_pkgs); i++) {
+          bids.pop_back();
+        }
```

Comment:
May be more straightforward
```suggestion
        bids.erase(bids.begin() + shippable_pkgs, bids.end())
```